spec/weak-types.html

```diff
@@ -0,0 +1,348 @@
+    </emu-clause>
+
+    <emu-clause id="sec-unforgeable-values">
+      <h1><ins>Unforgeable values</ins></h1>
+
+      <p>
+        <ins>A value _V_ is <dfn>unforgeable</dfn> if either one of the following conditions is true:</ins>
+        <ul>
+          <li><ins>it is an Object;</ins></li>
+          <li><ins>it is a Box whose [[Value]] internal slot is an unforgeable value;</ins></li>
+          <li><ins>it is a Tuple whose [[Sequence]] internal slot contains at least one unforgeable value;</ins></li>
+          <li><ins>it is a Record whose [[Fields]] internal slot contains at least one { [[Key]], [[Value]] } pair such that [[Value]] is an unforgeable value.</ins></li>
+        </ul>
+      </p>
+
+      <emu-note>
+        <p><ins>Symbols created by the `Symbol()` constructor are technically unforgeable: you cannot recreate one with the same identity. However, they are not covered by this definition and they are not allowed as `WeakMap` keys, `WeakSet` values, or as `WeakRef` and `FinalizationRegistry` targets.</ins></p>
+      </emu-note>
+
+      <p>
+        <ins>An unforgeable value _V_ contains an object _o_ if either one of the following conditions is true:</ins>
+        <ul>
+          <li><ins>_V_ is _o_;</ins></li>
+          <li><ins>_V_ is a Box whose [[Value]] internal slot contains _o_;</ins></li>
+          <li><ins>_V_ is a Tuple whose [[Sequence]] internal slot contains at least one unforgeable value that contains _o_;</ins></li>
+          <li><ins>_V_ is a Record whose [[Fields]] internal slot contains at least one { [[Key]], [[Value]] } pair such that [[Value]] is an unforgeable value that contains _o_.</ins></li>
+        </ul>
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-liveness">
+      <h1>Liveness</h1>
+
+      <p>For some set of objects _S_, a <dfn>hypothetical WeakRef-oblivious</dfn> execution with respect to _S_ is an execution whereby the abstract operation WeakRefDeref of a WeakRef whose referent is <ins>an unforgeable value that <emu-xref href="#sec-unforgeable-values">contains</emu-xref></ins> an element of _S_ always returns *undefined*.</p>
+
+      <emu-note>
+        WeakRef-obliviousness, together with liveness, capture two notions. One, that a WeakRef itself does not keep an <del>object</del><ins>unforgeable value</ins> alive. Two, that cycles in liveness does not imply that an <del>object</del><ins>unforgeable value</ins> is live. To be concrete, if determining _obj_'s liveness depends on determining the liveness of another WeakRef referent, <del>_obj2_</del><ins>_val2_</ins>, <del>_obj2's liveness</del><ins>the liveness of the objects _obj2_ such that _val2_ <emu-xref href="#sec-unforgeable-values">contains</emu-xref> _obj2_</ins> cannot assume _obj_'s liveness, which would be circular reasoning.
```

Comment:
Same here, I think we can simplify by replacing all objects occurrences by unforgeable value, without touching the concept of "contains"

```suggestion
        WeakRef-obliviousness, together with liveness, capture two notions. One, that a WeakRef itself does not keep an <del>object</del><ins>unforgeable value</ins> alive. Two, that cycles in liveness does not imply that an <del>object</del><ins>unforgeable value</ins> is live. To be concrete, if determining <ins>_val_</ins><del>_obj_</del>'s liveness depends on determining the liveness of another WeakRef referent, <del>_obj2_</del><ins>_val2_</ins>, <del>_obj2_</del><ins>_val2_</ins>'s liveness cannot assume <ins>_val_</ins><del>_obj_</del>'s liveness, which would be circular reasoning.
```